libmisc/copydir.c

```diff
@@ -719,11 +718,10 @@ static int copy_special (const struct path_info *src, const struct path_info *ds
 	if (   (mknodat (dst->dirfd, dst->name, statp->st_mode & ~07777U, statp->st_rdev) != 0)
```

Comment:
Why not a mode of 0600 here?

Reply:
I'm also missing a more restrictive mode here.

Reply:
The mode is already very restrictive. Please not that the mask is inverted 07777. Only the highest bits are preserved which are required for mknodat to figure out the actual type. See its manual page for more details.